File: C++/0045-jump-game-ii.cpp
class Solution {
 public:
  int jump(vector<int>& nums) {
    int jumps = 0, currentJumpEnd = 0, farthest = 0;
    for (int i = 0; i < nums.size() - 1; i++) {
      farthest = max(farthest, i + nums[i]);
      if (i == currentJumpEnd) {
        jumps++;
        currentJumpEnd = farthest;
      }
    }
    return jumps;
  }
};

File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return 0
        dp = [0]
        for i in range(len(nums)):
            dp.append(max((nums[j] + j for j in range(dp[i] + 1))))
            if dp[i + 1] >= len(nums) - 1:
                break
        return len(dp) - 1